autosubmit/helpers/utils.py

```diff
@@ -220,3 +221,31 @@ def strtobool(val):
         return 0
     else:
         raise ValueError("invalid truth value %r" % (val,))
+
+
+def get_rc_path(machine: bool, local: bool) -> Path:
+    """Get the ``.autosubmit.rc`` path.
+
+    If the environment variable ``AUTOSUBMIT_CONFIGURATION`` is specified in the
+    system, this function will return a ``Path`` pointing to that value.
+
+    If ``machine`` is ``True``, it will use the file from ``/etc/.autosubmitrc``
+    (pay attention to the dot prefix).
+
+    Else, if ``local`` is ``True``, it will use the file from  ``./.autosubmitrc``
+    (i.e. it will use the current working directory for the process).
+
+    Otherwise, it will load the file from ``~/.autosubmitrc``, for the user
+    currently running Autosubmit.
+    """
+    if 'AUTOSUBMIT_CONFIGURATION' in os.environ:
+        return Path(os.environ['AUTOSUBMIT_CONFIGURATION'])
+
+    if machine:
+        rc_path = '/etc'
+    elif local:
+        rc_path = '.'
+    else:
+        rc_path = Path.home()
+
+    return Path(rc_path) / '.autosubmitrc'
```

Comment:
Moved the logic from `autosubmit/autosubmit.py` to this smaller function to make it easier to test & reuse (e.g. in tests/fixtures).